test/X86/bug-reorder-bb-jrcxz.s

```diff
@@ -0,0 +1,597 @@
+# The test performs a BB reordering with unsupported 
+# instruction jrcxz. Reordering works correctly with the
+# following options: None, Normal or Reverse. Other strategies 
+# are completed with Assertion `isIntN(Size * 8 + 1, Value).
+# The root cause is the distance between BB where one contains 
+# jrcxz instruction.
+# Example: OpenSSL
+# https://github.com/openssl/openssl/blob/master/crypto/bn/asm/x86_64-mont5.pl#L3319
+
+# REQUIRES: system-linux
+
+# RUN: llvm-mc -filetype=obj -triple x86_64-unknown-unknown \
+# RUN:   %s -o %t.o
+# RUN: link_fdata %s %t.o %t.fdata
+# RUN: %host_cc --version
+# RUN: %host_cc %t.o -falign-labels -march=native -o %t.exe -Wl,-q
+
+# RUN:  llvm-bolt %t.exe -o %t.bolted -data %t.fdata \
+# RUN:    -debug -print-disasm -print-all -v=2 -lite=false \
+# RUN:    -reorder-blocks=cache+ -reorder-functions=hfsort \
+# RUN:    -split-functions=2 -split-all-cold -split-eh -dyno-stats \
+# RUN:    -print-finalized -max-funcs=200 | FileCheck %s
+
+# CHECK: const: Assertion `isIntN(Size * 8 + 1, Value) && \
```

Comment:
The check should check that it does not assert (that BOLT is indeed fixed). You can do that with "CHECK-NOT", and check that old bolt fails the test, but new bolt passes the test.

But I would rather not check an assertion (because the build can be done without assertions enabled). So it's better to just check that BOLT finishes without issues (don't check anything, and if BOLT crashes, the test will faill because not command line in RUN: is supposed to return non-zero exit code).